tokio-executor/src/typed.rs

```diff
@@ -0,0 +1,26 @@
+use SpawnError;
+
+/// TODO: DOX
+pub trait TypedExecutor<T> {
+
+    /// TODO: DOX
+    fn typed_spawn(&mut self, future: T) -> Result<(), SpawnError>;
```

Comment:
I have a minor preference for naming this method `spawn`.

Reply:
I opted for `typed_spawn` to avoid name conflicts since most types that impl `Executor` will also implement `TypedExecutor`. Still prefer `spawn`?

Reply:
I figured that's why you did, but I expect only 1 trait would be imported, and so `typed_spawn` reads weird.

Reply:
I'm ok w/ switching it to `spawn`.